src/main/java/net/fabricmc/loom/mixin/MixinMappingWriterTiny.java

```diff
@@ -46,20 +51,53 @@ public MixinMappingWriterTiny(Messager messager, Filer filer) {
 	@Override
 	public void write(String output, ObfuscationType type, IMappingConsumer.MappingSet<MappingField> fields, IMappingConsumer.MappingSet<MappingMethod> methods) {
 		if (output != null) {
-			String from = type.getKey().split(":")[0];
-			String to = type.getKey().split(":")[1];
+			String[] parts = type.getKey().split(":");
+			String from = parts[0];
+			String to = parts[1];
+
+			Map<String, Collection<String>> classesData = new TreeMap<>(); // Sorted to make 
+
+			print(classesData, "f", fields);
+			print(classesData, "m", methods);
 
 			try (PrintWriter writer = this.openFileWriter(output, type + " output TinyMappings")) {
-				writer.println(String.format("v1\t%s\t%s", from, to));
-				for (IMappingConsumer.MappingSet.Pair<MappingField> pair : fields) {
-					writer.println(String.format("FIELD\t%s\t%s\t%s\t%s", pair.from.getOwner(), pair.from.getDesc(), pair.from.getSimpleName(), pair.to.getSimpleName()));
-				}
-				for (IMappingConsumer.MappingSet.Pair<MappingMethod> pair : methods) {
-					writer.println(String.format("METHOD\t%s\t%s\t%s\t%s", pair.from.getOwner(), pair.from.getDesc(), pair.from.getSimpleName(), pair.to.getSimpleName()));
+				writer.println(String.format("tiny\t2\t0\t%s\t%s", from, to));
+				// writer.println("\tsorted-classes"); todo write sorted attribute
+				for (Collection<String> lines : classesData.values()) {
+					for (String line : lines) {
+						writer.println(line);
+					}
 				}
 			} catch (IOException e) {
 				e.printStackTrace();
 			}
 		}
 	}
+
+	private static void print(Map<String, Collection<String>> classesData, String type, IMappingConsumer.MappingSet<? extends IMapping<?>> mappingSet) {
+		for (IMappingConsumer.MappingSet.Pair<? extends IMapping<?>> pair : mappingSet) {
+			classesData.computeIfAbsent(pair.from.getOwner(), key -> {
+				Collection<String> ret = new ArrayList<>();
+				ret.add(printClassLine(key, pair.to.getOwner()));
+				return ret;
+			}).add(printClassMemberLine(type, pair.from.getDesc(), pair.from.getName(), pair.to.getName()));
+		}
+	}
+
+	private static String printClassLine(String oldName, String newName) {
```

Comment:
These 2 return `String` so they should be named  `create...` or similar  instead of `print...`.

Reply:
this is not creating things; this is the printed line for a class

Reply:
Methods that return `void` shouldn't look at same as those that return `String`.

Reply:
this is impl detail, no one cares

Reply:
I don't think that high code quality should be restricted to api methods.

Reply:
sure I will name this `a` so that the class file will get smaller. High quality code lol

Reply:
No I don't think that if the class file is smaller it would be higher quality code. But if the code is more readable then it is higher quality...